openedx/core/djangoapps/external_user_ids/models.py

```diff
@@ -103,3 +103,58 @@ def add_new_user_id(cls, user, type_name):
                 )
             )
         return external_id, created
+
+    @classmethod
+    def batch_get_or_create_user_ids(cls, users, type_name):
+        """
+        Creates ExternalIds in batch.
+
+        Given a list of users and a type_name, this method creates new external ids for
+        users that do not already have one. Then returns a dictionary mapping user id with
+        corresponding external id.
+
+        Arguments:
+            users: List of User to create the IDs for
+            type_name (str): Name of the type of ExternalId
+        Returns:
+            dict: Returns ExternalIds mapped by User.id
+                {
+                    user_id: ExternalId
+                }
+        """
+        try:
+            type_obj = ExternalIdType.objects.get(name=type_name)
+        except ExternalIdType.DoesNotExist:
+            LOGGER.info(
+                'Batch ID Creation failed, no external id type of {type}'.format(
+                    type=type_name
+                )
+            )
+            return None
```

Comment:
Should this be `{}` instead to stick to the signature of the function?

Reply:
Or would it be better to raise an exception? I don't know about ExternalIdType, and what callers expect.

Reply:
(or at least update the docstring)

Reply:
The reasoning behind sending `None` is to keep it similar to `add_new_user_id` which also returns `None` if fails. I've updated the docstring.